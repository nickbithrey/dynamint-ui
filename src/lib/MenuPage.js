import React from 'react';
import { Route, Switch } from 'react-router-dom';

const MenuPage = ({routes}) => (
	<div>
		<Switch>
			{mapRoutes(routes)}
		</Switch>
	</div>
);

function mapRoutes(routes) {
	return routes.map(mapRoute);
}

function mapRoute(route) {
	return <Route key={route.path} exact={route.exact} path={route.path} component={route.component} />
}

export default MenuPage;