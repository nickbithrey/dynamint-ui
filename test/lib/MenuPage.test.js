import React from 'react';
import { shallow } from 'enzyme';

import MenuPage from '~/lib/MenuPage';

const Selection = () => <h1>Selection</h1>;
const Details = () => <h1>Details</h1>;

describe('<MenuPage /> rendering', () => {

	it('matches snapshot', () => {
		const routes = [
			{exact: true, path: '/compconfig', component: Selection},
			{exact: false, path: '/compconfig/:type', component: Details}
		];
		const wrapper = shallow( <MenuPage routes={routes} /> );
		
		expect(wrapper).toMatchSnapshot();
	});
	
	it('will render the full set of routes', () => {
		const routes = [
			{exact: true, path: '/compconfig', component: Selection},
			{exact: false, path: '/compconfig/:type', component: Details}
		];
		const wrapper = shallow( <MenuPage routes={routes} /> );

		const switchWrapper = wrapper.find('Switch');
		expect(switchWrapper).toHaveLength(1);
		const routeWrappers = switchWrapper.find('Route');
		expect(routeWrappers).toHaveLength(routes.length);
		routes.forEach((r,i) => {
			expect(routeWrappers.at(i).prop('path')).toBe(r.path);
			expect(routeWrappers.at(i).prop('exact')).toBe(r.exact);
			expect(routeWrappers.at(i).prop('component')).toBe(r.component);
		})
	});
		
});
	
