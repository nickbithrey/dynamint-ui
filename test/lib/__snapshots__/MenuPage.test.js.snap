// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuPage /> rendering matches snapshot 1`] = `
<div>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      key="/compconfig"
      path="/compconfig"
    />
    <Route
      component={[Function]}
      exact={false}
      key="/compconfig/:type"
      path="/compconfig/:type"
    />
  </Switch>
</div>
`;
